2024-09-26-typescript-effect/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import { Effect, Array, Option, pipe, Stream, Chunk, StreamEmit } from "effect";
+import { Schema } from "@effect/schema";
+import {
+  FetchHttpClient,
+  HttpClient,
+  HttpClientResponse,
+} from "@effect/platform";
+
+const CityResponse = Schema.Struct({
+  name: Schema.String,
+  country_code: Schema.String.pipe(Schema.length(2)),
+  latitude: Schema.Number,
+  longitude: Schema.Number,
+});
+
+type CityResult = Schema.Schema.Type<typeof CityResponse>;
+
+const GeocodingResponse = Schema.Struct({
+  results: Schema.Array(CityResponse),
+});
+
+type GeocodingResponse = Schema.Schema.Type<typeof GeocodingResponse>;
+
+const Weather = Schema.Struct({
+  current_units: Schema.Struct({
+    temperature_2m: Schema.String,
+    relative_humidity_2m: Schema.String,
+    apparent_temperature: Schema.String,
+    precipitation: Schema.String,
+  }),
+  current: Schema.Struct({
+    temperature_2m: Schema.Number,
+    relative_humidity_2m: Schema.Number,
+    apparent_temperature: Schema.Number,
+    precipitation: Schema.Number,
+  }),
+});
+
+type Weather = Schema.Schema.Type<typeof Weather>;
+
+// The field input
+const city = Option.fromNullable(
```

Comment:
Can you pls use the naming convention from the blogpost? cityElement, citiesElement, etc.

Reply:
Done